Libraries/src/Amazon.Lambda.AspNetCoreServer.Hosting/Internal/LambdaSnapstartExecuteRequestsBeforeSnapshotHelper.cs

```diff
@@ -0,0 +1,220 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using Amazon.Lambda.APIGatewayEvents;
+using Amazon.Lambda.ApplicationLoadBalancerEvents;
+using Amazon.Lambda.RuntimeSupport;
+using Amazon.Lambda.RuntimeSupport.Helpers;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Amazon.Lambda.AspNetCoreServer.Hosting.Internal;
+
+/// <summary>
+/// Contains the plumbing to register a user provided <see cref="Func{HttpClient, Task}"/> inside
+/// <see cref="Amazon.Lambda.Core.SnapshotRestore.RegisterBeforeSnapshot"/>.
+/// The function is meant to initialize the asp.net and lambda pipelines during
+/// <see cref="Core.SnapshotRestore.RegisterBeforeSnapshot"/> and improve the
+/// performance gains offered by SnapStart.
+/// <para />
+/// It works by construction a specialized <see cref="HttpClient" /> that will intercept requests
+/// and saved them inside <see cref="LambdaSnapstartInitializerHttpMessageHandler.CapturedHttpRequests" />.
+/// <para /> 
+/// Intercepted requests are then be processed later by <see cref="SnapstartHelperLambdaRequests.ExecuteSnapstartInitRequests"/>
+/// which will route them correctly through a simulated asp.net/lambda pipeline.
+/// </summary>
+internal class LambdaSnapstartExecuteRequestsBeforeSnapshotHelper
+{
+    private readonly LambdaEventSource _lambdaEventSource;
+
+    public LambdaSnapstartExecuteRequestsBeforeSnapshotHelper(LambdaEventSource lambdaEventSource)
+    {
+        _lambdaEventSource = lambdaEventSource;
+    }
+
+    /// <inheritdoc cref="RegisterInitializerRequests"/>
+    [RequiresUnreferencedCode("Serializes object to json")]
```

Comment:
You should be able to do this without having to set the `RequiresUnreferencedCode`. You are dealing with a few small known types that are serialized. You can define a JSON serialization context and use that.

Reply:
Wasn't sure how to update json serializer to be aot compliant - followed pattern elsewhere in the repo and removed `RequiresUnreferencedCode`